jsonrpc/codec.go

```diff
@@ -12,13 +12,20 @@ import (
 type Request struct {
 	ID     interface{}     `json:"id"`
 	Method string          `json:"method"`
-	Params json.RawMessage `json:"params"`
+	Params json.RawMessage `json:"params,omitempty"`
+}
+
+// ErrorResponse is a jsonrpc error response
+type ErrorResponse struct {
+	JSONRPC string       `json:"jsonrpc"`
+	ID      interface{}  `json:"id,omitempty"`
+	Error   *ErrorObject `json:"error"`
 }
 
 // Response is a jsonrpc response
 type Response struct {
-	ID      interface{}     `json:"id"`
 	JSONRPC string          `json:"jsonrpc"`
+	ID      interface{}     `json:"id"`
```

Comment:
I know this was not modified as part of this task, but we should really consider if it is worthwhile to have the `ID` field as type interface.

As far as I can remember, this field can be either a number or a string (uuid).

Reply:
currently, i am sticking to interface, i will open a new task for this specially